src/models/piranha_arguments.rs

```diff
@@ -61,172 +68,237 @@ pub struct PiranhaArguments {
+        "number_of_ancestors_in_parent_scope",
+        default_number_of_ancestors_in_parent_scope,
+        "num"
       ),
-      substitutions: Self::_merge(
-        self.substitutions.clone(),
-        other.substitutions,
-        default_substitutions(),
+      delete_consecutive_new_lines: get_keyword_arg!(
+        "delete_consecutive_new_lines",
+        default_delete_consecutive_new_lines,
+        "bool"
       ),
-      path_to_configurations: Self::_merge(
-        self.path_to_configurations.clone(),
-        other.path_to_configurations,
-        default_path_to_configurations(),
+      global_tag_prefix: get_keyword_arg!("global_tag_prefix", default_global_tag_prefix, "string"),
+      delete_file_if_empty: get_keyword_arg!(
+        "delete_file_if_empty",
+        default_delete_file_if_empty,
+        "bool"
       ),
-      path_to_output_summary: Self::_merge(
-        self.path_to_output_summary.clone(),
-        other.path_to_output_summary,
-        default_path_to_output_summaries(),
+      path_to_output_summary: get_keyword_arg!(
+        "path_to_output_summary",
+        default_path_to_output_summaries,
+        "option"
       ),
-      language: Self::_merge(self.language.clone(), other.language, default_languages()),
-      piranha_language: Self::_merge(
-        self.piranha_language.clone(),
-        other.piranha_language,
-        default_piranha_language(),
-      ),
-      delete_file_if_empty: Self::_merge(
-        self.delete_file_if_empty,
-        other.delete_file_if_empty,
-        default_delete_file_if_empty(),
-      ),
-      delete_consecutive_new_lines: Self::_merge(
-        self.delete_consecutive_new_lines,
-        other.delete_consecutive_new_lines,
-        default_delete_consecutive_new_lines(),
-      ),
-      global_tag_prefix: Self::_merge(
-        self.global_tag_prefix.clone(),
-        other.global_tag_prefix,
-        default_global_tag_prefix(),
-      ),
-      number_of_ancestors_in_parent_scope: Self::_merge(
-        self.number_of_ancestors_in_parent_scope,
-        other.number_of_ancestors_in_parent_scope,
-        default_number_of_ancestors_in_parent_scope(),
-      ),
-      cleanup_comments_buffer: Self::_merge(
-        self.cleanup_comments_buffer,
-        other.cleanup_comments_buffer,
-        default_cleanup_comments_buffer(),
+      ..Default::default()
+    }
+    .merge(PiranhaArgumentsBuilder::default().build())
+  }
+}
+
+impl PiranhaArguments {
+  pub fn get_language(&self) -> String {
+    self.language.clone()
+  }
+
+  pub fn merge(&self, other: PiranhaArguments) -> Self {
+    /// Accepts field name (e.g. `language`) and function name (e.g. `default_language`) which returns default value for that field.
+    /// It checks if the value `self.language` is same as value returned by `default_language()`.
+    /// If the value is same it returns the value from `other (e.g. `other.language`) else it returns it from `self`
```

Comment:
So, the default is the same explicitly setting this to `java`. Does it mean that when merging two instances of `PiranhaArguments`, one of which has `language = "java"` explicitly and the other some other language, the non-java one will be preferred? When would we merge two sets of arguments, specially for two separate languages?

Reply:
We will merge them, once we start supporting `language: Vec<String>` (look at my comment above)